src/Compilers/Core/Portable/SourceGeneration/GeneratorContexts.cs

```diff
@@ -15,6 +15,7 @@ namespace Microsoft.CodeAnalysis
     /// <summary>
     /// Context passed to a source generator when <see cref="ISourceGenerator.Execute(GeneratorExecutionContext)"/> is called
     /// </summary>
+    [Obsolete(SourceGenerationMessages.UseIncrementalGenerator)]
```

Comment:
Unfortunately, it seems that this is an older version of `ObsoleteAttribute` which doesn't have that property. But it would be great if that could be used, I think it could be used to suppress the warning globally across Roslyn.

Reply:
@jjonescz I think we can define our own `ObsoleteAttribute` as follows:

```csharp
namespace System
{
    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Struct | AttributeTargets.Enum |
        AttributeTargets.Interface | AttributeTargets.Constructor | AttributeTargets.Method | AttributeTargets.Property | AttributeTargets.Field | AttributeTargets.Event | AttributeTargets.Delegate,
        Inherited = false)]
    internal sealed class ObsoleteAttribute : Attribute
    {
        public ObsoleteAttribute()
        {
        }

        public ObsoleteAttribute(string? message)
        {
            Message = message;
        }

        public ObsoleteAttribute(string? message, bool error)
        {
            Message = message;
            IsError = error;
        }

        public string? Message { get; }

        public bool IsError { get; }

        public string? DiagnosticId { get; set; }

        public string? UrlFormat { get; set; }
    }
}
```

and apply it `[Obsolete(SourceGenerationMessages.UseIncrementalGenerator, DiagnosticId = "ID goes here")]`

Reply:
Thanks @Youssef1313, that looks good, I tried it. But declaring this type creates ambiguity with the original which is especially problematic in Visual Basic projects (which see this type due to `InternalsVisibleTo`) - they emit error BC30560 which cannot be suppressed.

Reply:
One solution I can think of is declaring `ObsoleteAttribute` with `file` modifier. Though since we are obsoleting APIs in different files, the code will have to be repeated.

Reply:
Seems that such type is not picked up by the compiler - it doesn't generate obsolete warnings.

Reply:
![image](https://user-images.githubusercontent.com/31348972/213225530-51f17f6c-df90-4f09-a227-f1fa809ad179.png)

Strange. It's picked up for me in a new netcoreapp3.1 app

Reply:
Maybe because you are in one file?

Reply:
Oh, no, I had a bug in my `ObsoleteAttribute` definition (introduced while playing with it). It indeed works with `file` modifier. Thanks.

Reply:
Except it seems it doesn't work across library boundaries.